es6-shim.js

```diff
@@ -665,7 +665,7 @@
   var addIterator = function (prototype, impl) {
     var implementation = impl || function iterator() { return this; };
     defineProperty(prototype, $iterator$, implementation);
-    if (!prototype[$iterator$] && Type.symbol($iterator$)) {
+    if (prototype && !prototype[$iterator$] && Type.symbol($iterator$)) {
```

Comment:
similarly here, i'd rather throw an exception when prototype is falsy if we need the check; otherwise can we revert it?

Reply:
I think it is okay to revert this.